modules/redis/redis-worker.c

```diff
@@ -0,0 +1,307 @@
+        }
+      else if (r->type == REDIS_REPLY_STRING || r->type == REDIS_REPLY_STATUS || r->type == REDIS_REPLY_ERROR)
+        {
+          msg_trace("REDIS command reply",
+                    evt_tag_str("str", r->str));
+        }
+      else
+        {
+          msg_trace("REDIS command unhandled reply",
+                    evt_tag_int("type", r->type));
+        }
+    }
+}
+
+
+
+
+static gboolean
+send_redis_command(RedisDestWorker *self, const char *format, ...)
+{
+  va_list ap;
+  va_start(ap, format);
+  redisReply *reply = redisvCommand(self->c, format, ap);
+  va_end(ap);
+
+  gboolean retval = reply && (reply->type != REDIS_REPLY_ERROR);
+  if (reply)
+    {
+      _trace_reply_message(reply);
+      freeReplyObject(reply);
+    }
+  return retval;
+}
+
+static gboolean
+check_connection_to_redis(RedisDestWorker *self)
+{
+  return send_redis_command(self, "ping");
+}
+
+static gboolean
+authenticate_to_redis(RedisDestWorker *self, const gchar *password)
+{
+  return send_redis_command(self, "AUTH %s", ((RedisDriver *) self->super.owner)->auth);
+}
+
+static gboolean
+redis_dd_connect(LogThreadedDestWorker *s)
+{
+  RedisDestWorker *self = (RedisDestWorker *)s;
+  RedisDriver *owner = (RedisDriver *) self->super.owner;
+
+  if (self->c && check_connection_to_redis(self))
+    return TRUE;
+
+  if (self->c)
+    redisFree(self->c);
+
+  self->c = redisConnectWithTimeout(owner->host, owner->port, self->timeout);
+
+  if (self->c == NULL || self->c->err)
+    {
+      if (self->c)
+        {
+          msg_error("REDIS server error during connection",
+                    evt_tag_str("driver", owner->super.super.super.id),
+                    evt_tag_str("error", self->c->errstr),
+                    evt_tag_int("time_reopen", self->super.time_reopen));
+        }
+      else
+        {
+          msg_error("REDIS server can't allocate redis context");
+        }
+      return FALSE;
+    }
+
+  if (owner->auth)
+    if (!authenticate_to_redis(self, owner->auth))
+      {
+        msg_error("REDIS: failed to authenticate",
+                  evt_tag_str("driver", owner->super.super.super.id));
+        return FALSE;
+      }
+
+  if (!check_connection_to_redis(self))
+    {
+      msg_error("REDIS: failed to connect",
+                evt_tag_str("driver", owner->super.super.super.id));
+      return FALSE;
+    }
+
+  if (self->c->err)
+    return FALSE;
+
+  msg_debug("Connecting to REDIS succeeded",
+            evt_tag_str("driver", owner->super.super.super.id));
+
+  return TRUE;
+}
+
+
+LogThreadedDestWorker *redis_dw_new(LogThreadedDestDriver *owner, gint worker_index)
+{
+  RedisDestWorker *self = g_new0(RedisDestWorker, 1);
+
+  log_threaded_dest_worker_init_instance(&self->super, owner, worker_index);
+
+  self->super.thread_init = redis_worker_thread_init;
+  self->super.thread_deinit = redis_worker_thread_deinit;
+  self->super.connect = redis_dd_connect;
+  self->super.disconnect = redis_dd_disconnect;
```

Comment:
The `dd` usually refers to destination driver, and the `dw` a destination worker. Please use the proper naming, in this case `redis_dw_disconnect` or the `redis_worker_disconnect` is fine. But use only either (`worker`/`dw`) for all method in this compilation unit.